sql/src/main/java/org/apache/druid/sql/calcite/expression/Expressions.java

```diff
@@ -242,6 +242,7 @@ private static DruidExpression rexCallToDruidExpression(
                                                            .lookupOperatorConversion(operator);
 
     if (conversion == null) {
+      plannerContext.setPlanningError("SQL query requires '%s' operator that is not supported.", operator.getName());
```

Comment:
Since this method is mostly used as a helper/util method, setting the error here might not be desirable in every use case. Would it be better if it is moved to a place like this?  [Projections, line 105](https://github.com/apache/druid/blob/476d0bf4be4199e97695bd568d165cda98523d37/sql/src/main/java/org/apache/druid/sql/calcite/rel/Projection.java#L105)

Reply:
the absence of a corresponding operator seems like a genuine feature gap. Also, I can populate a more precise error here than I can in the uppar part of the call stack. Since a null can be returned for other reasons.